### PR TITLE
[CP-2912] Mapper for Outlook contacts

### PR DESCRIPTION
**Pull request by MateuszMudita**
JIRA Reference: [CP-2912](https://appnroll.atlassian.net/browse/CP-2912?atlOrigin=eyJpIjoiNGFjODc5M2E3ZDYzNDBhY2EyMWQyNDE5NWUxY2Q2OTYiLCJwIjoiaiJ9)

### :memo: Description ️

- Created mapper file for Outlook contacts

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2912]: https://appnroll.atlassian.net/browse/CP-2912?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { UnifiedContact } from "device/models"
+import { Contact } from "Core/contacts/reducers/contacts.interface"
+import { getDisplayName } from "../helpers"
+// import { getDisplayName } from "../helpers"
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:9]:
```suggestion
```

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        id: contact.id,
+        firstName: contact.firstName,
+        lastName: contact.lastName,
+        phoneNumbers: [
+          ...(contact.primaryPhoneNumber
+            ? [
+                {
+                  value: contact.primaryPhoneNumber,
+                  preference: 1,
+                  type: "other",
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:25]: I think that you can remove "other" type, the field is optional in the interface
```suggestion
```

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                },
+              ]
+            : []),
+          ...(contact.secondaryPhoneNumber
+            ? [
+                {
+                  value: contact.secondaryPhoneNumber,
+                  preference: contact.primaryPhoneNumber ? 2 : 1,
+                  type: "other",
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:34]:
```suggestion
```

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+              ]
+            : []),
+        ],
+        emailAddresses: [
+          ...(contact.email
+            ? [
+                {
+                  value: contact.email,
+                  preference: 1,
+                  type: "other",
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:45]:
```suggestion
```

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          ...(contact.firstAddressLine || contact.secondAddressLine
+            ? [
+                {
+                  streetAddress:
+                    contact.firstAddressLine || contact.secondAddressLine,
+                  ...(contact.firstAddressLine &&
+                    contact.secondAddressLine && {
+                      extendedAddress: contact.secondAddressLine,
+                    }),
+                  type: "other",
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:60]:
```suggestion
```

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                },
+              ]
+            : []),
+        ],
+        note: contact.note,
+      } as UnifiedContact
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:66]: 🟡 
Please try to handle all required properties of `UnifiedContact` without using the assert type 😉 
```suggestion
      }
```

**Review by dkarski — APPROVED** [review]

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { UnifiedContact } from "device/models"
+import { Contact } from "Core/contacts/reducers/contacts.interface"
+import { getDisplayName } from "../helpers"
+
+export const mapOutlookApi = (contacts: Contact[]): UnifiedContact[] => {
+  try {
```
Comment (lkowalczyk87) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:11]: I see that error in any of the contact entries replacing all contacts with an empty array. Maybe we can go in a different direction eg. corrupted contacts are skipped, so in a case when one out of 100 contacts is corrupted we still can process 99 correct contacts. TBD during standup meeting.

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+          ],
+          note: contact.note,
+          organizations: [],
+          urls: [],
+        }
+      } catch {
+        return null
+      }
+    })
+    .filter((contact): contact is UnifiedContact => contact != null)
```
Comment (lkowalczyk87) [inline libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:69]: `.filter(Boolean)` ;)

**Review by lkowalczyk87 — APPROVED** [review]

File: libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts (hunk 8 above)
Comment (mkurczewski) [reply libs/generic-view/store/src/lib/imports/contacts-mappers/outlook-api/map-outlook-api.ts:69]: > `.filter(Boolean)` ;)

More like 
```TSX
.filter(Boolean) as UnifiedContact[]
```

**Review by lkowalczyk87 — APPROVED** [review]

**Pull request merged**